pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -335,6 +335,10 @@ func (p *Pod) RestorePodSetsInfo(_ []podset.PodSetInfo) bool {
 // Finished means whether the job is completed/failed or not,
 // condition represents the workload finished condition.
 func (p *Pod) Finished() (message string, success, finished bool) {
+	if p.isServing() {
```

Comment:
I think this is a bugfix for an issue on its own. Let me open so that we can fix issues one-by-one.